### PR TITLE
[IOS-388, IOS-389, IOS-390]: Initial group creation

### PR DESCRIPTION
**Pull request by yuliaveres**
Initial group chat support.
(Contains not merged avatar related changes as well, was branched out from https://github.com/toshiapp/toshi-ios-client/pull/582).
Uses hardcoded default recipients addresses until users picker is on a branch.

Does not contain group membership request confirmation due to lack of design at the moment.

img_0036

Also no design for status messages YET ("group created" etc)

### CODE REVIEW

File: circle.yml [hunk 1]
```diff
@@ -5,7 +5,7 @@ machine:
 test:
   test:
   override:
-    - set -o pipefail && xcodebuild test -workspace Toshi.xcworkspace -scheme "Tests" -destination 'platform=iOS Simulator,name=iPhone 7' | xcpretty --color && exit ${PIPESTATUS[0]}
+    - set -o pipefail && xcodebuild clean build test -workspace Toshi.xcworkspace -scheme "Tests" -destination 'platform=iOS Simulator,name=iPhone 7' | xcpretty --color && exit ${PIPESTATUS[0]}
```
Comment (yuliaveres) [inline circle.yml]: good attempt ! 😃  thank you

File: Toshi/Models/Notifications/SignalNotificationManager.swift [hunk 2]
```diff
@@ -18,9 +18,7 @@ import UIKit
 public class SignalNotificationManager: NSObject, NotificationsProtocol {
 
     static var tabbarController: TabBarController? {
```
Comment (roberthein) [inline Toshi/Models/Notifications/SignalNotificationManager.swift]: Can we remove this static var completely?

File: Toshi/MessagesUI/Cells/MessagesBasicCell.swift [hunk 3]
```diff
@@ -95,6 +100,18 @@ class MessagesBasicCell: UITableViewCell {
     private var contentLayoutGuideTopConstraint: NSLayoutConstraint?
     private var bottomLayoutGuideHeightConstraint: NSLayoutConstraint?
 
+    private lazy var avatarTapGestureRecogniser: UITapGestureRecognizer = {
+        return UITapGestureRecognizer(target: self, action: #selector(didTapAvatarImageView(_:)))
+    }()
+
+    @objc private func didTapAvatarImageView(_ gesture: UITapGestureRecognizer) {
```
Comment (roberthein) [inline Toshi/MessagesUI/Cells/MessagesBasicCell.swift]: Can the `func` be with the `func`s?

File: Toshi/MessagesUI/Cells/MessagesBasicCell.swift [hunk 4]
```diff
@@ -116,7 +133,7 @@ class MessagesBasicCell: UITableViewCell {
             let isFirstMessage = positionType == .single || positionType == .top
             contentLayoutGuideTopConstraint?.constant = isFirstMessage ? 8 : 4
 
-            let isAvatarHidden = positionType == .top || positionType == .middle || isOutGoing
+            let isAvatarHidden = positionType == .middle || positionType == .top || isOutGoing
```
Comment (roberthein) [inline Toshi/MessagesUI/Cells/MessagesBasicCell.swift]: You didn't like the order? XD

**Pull request merged**